app/controllers/GuidanceController.scala

```diff
@@ -133,34 +131,40 @@ class GuidanceController @Inject() (
         case err @ Left(_) => err
       }
 
-    val sessionId: String = hc.sessionId.fold(java.util.UUID.randomUUID.toString)(_.value)
-    logger.info(s"Starting approval direct view with sessionId = $sessionId")
-    retreiveCacheAndRedirectToProcess(processId, sessionId, retrieveCacheAndRedirect(s"/$url"))
+    logger.info(s"Starting approval direct page view journey")
+    retreiveCacheAndRedirectToView(processId, retrieveCacheAndRedirect(s"/$url"))
   }
 
-  private def withSession[T](block: String => Future[RequestOutcome[T]])(implicit request: Request[_]): Future[RequestOutcome[T]] =
-    hc.sessionId.fold {
-      logger.error(s"Session Id missing from request when required")
-      Future.successful(Left(BadRequestError): RequestOutcome[T])
-    } { sessionId =>
-      logger.info(s"withSession, EG_SESSIONID ${request.session.get("EG_SESSION")}, sessionId $sessionId")
-      block(sessionId.value)
-    }
-
-  private def retreiveCacheAndRedirectToProcess(id: String, sessionId: String, processStartUrl: (String, String) => Future[RequestOutcome[String]])(
+  private def retreiveCacheAndRedirectToView(id: String, retreiveAndCache: (String, String) => Future[RequestOutcome[String]])(
```

Comment:
I think it is "retrieve" not "retreive". Personally, I'd split this method in two as to me caching the process and redirecting to another page are very different actions.